### PR TITLE
Move to API version 2019-03-14

### PR DESCRIPTION
**Pull request by remi-stripe**
The changes in the library itself look minor. But this API version changes the behavior of Subscriptions entirely under the hood. If the charge fails, you still get a Subscription back instead of an exception and it's on you to check the Subscription status.

While changing this, I also:
* removed `PaymentIntentLastError` in favor of `StripeError`
* forced the test suite to confirm we have the correct API version set in the library per stripe-mock.

Related to https://github.com/stripe/stripe-go/pull/811

r? @ob-stripe 
cc @stripe/api-libraries

### CODE REVIEW

**Comment by JoSaint13** [comment]
👍

**Comment by remi-stripe** [comment]
@ob-stripe Updated the PR, can you review tomorrow?

File: appveyor.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ install:
         7z.exe e -y -o".\stripe-mock" ".\stripe-mock\stripe-mock_${env:STRIPE_MOCK_VERSION}_windows_amd64.tar.gz" | Out-Null
         7z.exe x -y -o".\stripe-mock\stripe-mock_${env:STRIPE_MOCK_VERSION}" ".\stripe-mock\stripe-mock_${env:STRIPE_MOCK_VERSION}_windows_amd64.tar" | Out-Null
       }
-      $app = Start-Process -FilePath ".\stripe-mock\stripe-mock_${env:STRIPE_MOCK_VERSION}\stripe-mock.exe" -NoNewWindow -PassThru
+      $app = Start-Process -FilePath ".\stripe-mock\stripe-mock_${env:STRIPE_MOCK_VERSION}\stripe-mock.exe -strict-version-check" -NoNewWindow -PassThru
```
Comment (ob-stripe) [inline appveyor.yml]: Try this instead:
```suggestion
      $app = Start-Process -FilePath ".\stripe-mock\stripe-mock_${env:STRIPE_MOCK_VERSION}\stripe-mock.exe" -ArgumentList "-strict-version-check" -NoNewWindow -PassThru
```

**Comment by remi-stripe** [comment]
@ob-stripe PTAL

**Pull request merged**